### PR TITLE
MAINT: Ensure that meas_date=None round-trips if `anonymize_info` is used

### LINKED ISSUE #7067: meas_date is guessed during I/O when it is None
It seems that the `meas_date` is not saved / read in correctly. There seems to be a regression of issue #4949 with latest master. Here is the code to reproduce:

```py
import os.path as op

import mne
from mne.datasets import testing

data_path = testing.data_path()
raw_fname = op.join(data_path, 'MEG', 'sample',
                    'sample_audvis_trunc_raw.fif')

raw = mne.io.read_raw_fif(raw_fname)
raw.info['meas_date'] = None
raw.save('temp.fif')

raw2 = mne.io.read_raw_fif('temp.fif')
assert raw2.info['meas_date'] is None
```

I discovered this while working on https://github.com/mne-tools/mne-bids/pull/290/

### PR DESCRIPTION
**Pull request by larsoner**
First step hopefully in our grand plan.

Closes #7071
Closes #7067

### CODE REVIEW

File: mne/io/tests/test_meas_info.py [hunk 1]
```diff
@@ -245,6 +245,14 @@ def test_read_write_info(tmpdir):
     m2 = m2.hexdigest()
     assert m1 == m2
 
+    info = read_info(raw_fname)
+    info['meas_date'] = None
+    anonymize_info(info)
```
Comment (jasmainak) [inline mne/io/tests/test_meas_info.py]: is `info['meas_date'] = None` after this? According to https://mne.tools/dev/generated/mne.io.anonymize_info.html#mne.io.anonymize_info, isn't it going to be reset to Jan 1st 2000, no?

Comment (larsoner) [reply mne/io/tests/test_meas_info.py]: It's None, pushed a commit to assert it and updated docstring

**Comment by larsoner** [comment]
To help people transition, we could emit a warning when writing a file with `info['meas_date'] is None` but the other fields (`file_id`, `meas_id`) with values that will set it non-None after write-read round-trip. We can suggest that people do `anonymize_info` after setting `meas_date` to None. @jasmainak @agramfort WDYT?

**Comment by larsoner** [comment]
On second thought, we are in this release cycle going to make a `set_meas_date` function. People should use that going forward, so I think the warning will be unnecessary.

So this should be good for merge

**Comment by agramfort** [comment]
thx @larsoner !

**Pull request merged**